ultralytics/engine/exporter.py

```diff
@@ -452,19 +452,27 @@ def serialize(ov_model, file):
 
             def transform_fn(data_item):
                 """Quantization transform function."""
-                assert (
-                    data_item["img"].dtype == torch.uint8
-                ), "Input image must be uint8 for the quantization preprocessing"
-                im = data_item["img"].numpy().astype(np.float32) / 255.0  # uint8 to fp16/32 and 0 - 255 to 0.0 - 1.0
+                assert data_item.dtype == torch.uint8, "Input image must be uint8 for the quantization preprocessing"
+                im = data_item.numpy().astype(np.float32) / 255.0  # uint8 to fp16/32 and 0 - 255 to 0.0 - 1.0
                 return np.expand_dims(im, 0) if im.ndim == 3 else im
 
             # Generate calibration data for integer quantization
             LOGGER.info(f"{prefix} collecting INT8 calibration images from 'data={self.args.data}'")
-            data = check_det_dataset(self.args.data)
-            dataset = YOLODataset(data["val"], data=data, task=self.model.task, imgsz=self.imgsz[0], augment=False)
+            data = (
+                check_det_dataset(self.args.data) if self.args.task != "classify" else check_cls_dataset(self.args.data)
```

Comment:
@Burhan-Q in this kind of situation you don't want to repeat the argument, you can place the if else on the function instead of the function+argument, i.e.

data = (fcna if x else fcnb)(self.args.data)